core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -138,6 +141,8 @@ class KafkaServer(val config: KafkaConfig, time: Time = SystemTime, threadNamePr
   val correlationId: AtomicInteger = new AtomicInteger(0)
   val brokerMetaPropsFile = "meta.properties"
   val brokerMetadataCheckpoints = config.logDirs.map(logDir => (logDir, new BrokerMetadataCheckpoint(new File(logDir + File.separator +brokerMetaPropsFile)))).toMap
+  var clusterId: String = null
+  val clusterResourceListeners  = new ClusterResourceListeners()
```

Comment:
The new fields should be private. Maybe take the chance and change the existing ones to be private too.